### ISSUE TITLE
LIKE & GLOB operators

### ISSUE THREAD

**Issue by Hollow111** [tarantool/tarantool#3572]
LIKE & GLOB are operators for comparing left hand side predicand with the pattern. Currently their implementation & functionality is far from the one we want to have in Tarantool.

First of all GLOB is an SQLite extension and there's no GLOB in ANSI or almost any other DBMS.
(See "GLOB & LIKE" thread with @pgulutzan in @server-dev)
- [x] Remove GLOB support from Tarantool.

Next thing is that currently both LIKE and GLOB don't take collations into account (#3589), it only depends on what we use (LIKE is case insensitive, GLOB is case insensitive).
- [ ] Make comparison using LIKE depend on collation.
- [ ] Support explicit stmt collation type that has higher priority than column one.
- [ ] Remove "PRAGMA case_sensitive_like".

After #3251, #3334 commit function that compares pattern with row value predicand will only be supposed to work with UTF-8 strings.
- [x] Fix string comparison hanging. #3251, #3334.

Since there are "character like predicate" & "octet like predicate" for LIKE in ANSI SQL:
- [ ] Support byte-comparison (for BLOBs and probably byte string comparison hence it doesn't matter if we have invalid unicode symbols)

According to Unicode'\0' is a character (0000 NULL)
- [ ] Fix 'zero byte' for TEXT comparisons. #3542.

**Need feedback:**
- [ ] Prohibit anything other than BLOB & TEXT comparisons.

Unresolved questions:
1) What pairs of operand types should we support (for instance can it be byte comparison for 'BLOB LIKE TEXT')?

2) How exactly should we introduce 'binary value expression' for both pattern and row value predicand (X'someting' / 'TEXT' / hexadecimals / something_else)?

3) Should we maybe support UTF-16 and other comparisons?

**Note:**
LIKE operator mustn't use normalization (e.g.  'oe' LIKE  'œ' == 0).

**Comment by kyukhin** [comment]
1. IMHO, this is due to type derivation. I think we should ever allow comparing blob and string.
2. Need to look at other DBSMS/ANSI.
3. No.

**Comment by Totktonada** [comment]
1. I don't think we should support LIKE for anything except TEXT / VARCHAR / CHAR.

At least we should elaborate how and why one can want to use LIKE for BLOBs before discussing whether we should going to support them.

**Comment by Totktonada** [comment]
Remaining tasks:

* [#3589] Follow collations in LIKE operator.
* [#3523] Remove like() function.
* [#3954] Forbid anything except TEXT for LIKE operator.

I think this issue can be closed (as superseded by the issues above).

**Comment by kyukhin** [comment]
Superseded by issues in previous comment.
